File: Backend/Models/Irasas.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    [Table("irasas")]
    public class Irasas
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("id_dokumento")]
        public string IdDokumento { get; set; } = null!;

        [Required]
        [Column("pavadinimas")]
        public string Pavadinimas { get; set; } = null!;

        [Column("pradzia")]
        public DateTime Pradzia { get; set; }

        [Column("pabaiga")]
        public DateTime? Pabaiga { get; set; }
        
        [Column("statusas")]
        public string Statusas { get; set; } = null!;

        [JsonIgnore]
        public ICollection<TestasIrasas> Testai { get; set; } = new List<TestasIrasas>();
    }
}

File: frontend/lib/models/irasas.dart
class Irasas {
  final int id;
  String idDokumento;
  String pavadinimas;
  DateTime pradzia;
  DateTime? pabaiga;
  String statusas;
  int lokacijaId;

  Irasas({
    required this.id,
    required this.idDokumento,
    required this.pavadinimas,
    required this.pradzia,
    this.pabaiga,
    this.statusas = '',
    this.lokacijaId = 0,
  });

  factory Irasas.fromJson(Map<String, dynamic> j) => Irasas(
        id: j['id'] as int,
        idDokumento: j['idDokumento'] ?? j['id_dokumento'] ?? '',
        pavadinimas: j['pavadinimas'] ?? j['Pavadinimas'] ?? '',
      statusas: (j['statusas'] ?? j['Statusas'] ?? '') as String,
      lokacijaId: (j['lokacijaId'] ?? j['lokacija_id'] ?? 0) as int,
        pradzia: DateTime.parse(
          j['pradzia'] ?? j['Pradzia'] ?? DateTime.now().toIso8601String(),
        ).toLocal(),
        pabaiga: j['pabaiga'] == null
            ? null
            : DateTime.parse(j['pabaiga']).toLocal(),
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'idDokumento': idDokumento,
        'pavadinimas': pavadinimas,
        'pradzia': pradzia.toUtc().toIso8601String(),
        'pabaiga': pabaiga?.toUtc().toIso8601String(),
      'lokacijaId': lokacijaId,
      };
}
